rs/bitcoin/ckbtc/minter/src/updates/update_balance.rs

```diff
@@ -148,9 +149,16 @@ pub async fn update_balance<R: CanisterRuntime>(
         ic_cdk::trap("cannot update minter's balance");
     }
 
+    // Record start time of method execution for metrics
+    let start_time = runtime.time();
+
+    let observe_latency = |histogram: &mut LatencyHistogram| {
+        histogram.observe_latency_ns(runtime.time().saturating_sub(start_time))
+    };
```

Comment:
nit: this type of delayed callback I would avoid for readability reasons. I would favour having a kind of public method in `metrics.rs` similar to this [one](https://github.com/dfinity/ic/blob/80d6ecdc421047ef36acbbca5ea52cef0188acb9/rs/ethereum/cketh/minter/src/eth_rpc.rs#L909) doing something along the lines of:
```rust
    pub fn observe_latency(num_utxos: usize, start_ns: u64, end_ns: u64) {
    let duration = Duration::from_nanos(end_ns.saturating_sub(start_ns));
      UPDATE_CALL_LATENCY.with(|metrics| metrics.borrow_mut().observe_latency(num_utxos, duration));
    }
```

Like this your histogram logic only needs to deal with `Duration` which is usually less-error prone to handle